packages/@uppy/dashboard/src/utils/getDroppedFiles/index.js

```diff
@@ -0,0 +1,14 @@
+const webkitGetAsEntryApi = require('./utils/webkitGetAsEntryApi')
+const getFilesAndDirectoriesApi = require('./utils/getFilesAndDirectoriesApi')
+const fallbackApi = require('./utils/fallbackApi')
+
+module.exports = function getDroppedFiles (dataTransfer, callback) {
+  if (dataTransfer.items[0] && 'webkitGetAsEntry' in dataTransfer.items[0]) {
+    webkitGetAsEntryApi(dataTransfer, callback)
+  } else if ('getFilesAndDirectories' in dataTransfer) {
+    // Doesn't actually work in firefox, maybe in previous versions. webkitGetAsEntryApi() works.
+    getFilesAndDirectoriesApi(dataTransfer, callback)
+  } else {
+    fallbackApi(dataTransfer, callback)
+  }
+}
```

Comment:
I think we should move those util functions that work with drag-drop to @uppy/utils, so they can be shared (maybe later) with drag-drop plugin? Otherwise it will continue to depend on drag-drop module.

Reply:
@arturi, I agree, I even thought of factoring them out into a different library. Clean cross-browser folder drop implementation that's not tied to some html element seems to be lacking.
I'll move them to the @uppy/utils.